packages/rOpenScPCA/R/sweep-clusters.R

```diff
@@ -77,21 +96,25 @@ sweep_clusters <- function(
     stop("The first argument should be one of: a SingleCellExperiment object, a Seurat object, or a matrix with row names.")
   }
 
-  algorithm <- match.arg(algorithm)
-
   # Collect all specific inputs into a single list
-  # Even parameters that won't be used can be included
-  #   since calculate_clusters() will ignore them anyways
   sweep_params <- tidyr::expand_grid(
+    algorithm = unique(algorithm),
     weighting = unique(weighting),
     nn = unique(nn),
     resolution = unique(resolution),
     objective_function = unique(objective_function)
-  )
+  ) |>
+    # set unused parameters for the given algorithm to their defaults, since we can't provide NA
+    # to match.arg in calculate_clusters
+    dplyr::mutate(
+      resolution = ifelse(algorithm %in% c("louvain", "leiden"), resolution, 1),
+      objective_function = ifelse(algorithm == "leiden", objective_function, "CPM")
```

Comment:
I don't really like this because I think we want these values filled in table as NA (or absent from the table).  Can we leave it these NA and deal with them in the function? I would be willing to give up `match.arg` for this, but we probably can work around it too. Maybe something like:

```
if(is.na(objective_function)){
  objective_function = NULL
} else {
  objective_function = match.arg(objective_function)
}
```

Reply:
I confess, I have grown fond of `match.args`, so I'd rather have some ifs.